src/solution.jl

```diff
@@ -14,30 +14,33 @@ function DiscSolution(sol, semi, dimension_type::AbstractDimension)
 end
 
 #1D version of plot
-function dataplot(solution::DiscSolution{<:OneDimension}, t)
-    return PlotData1D(solution.sol(t), solution.semi)
+function dataplot(solution::DiscSolution{<:OneDimension}, solver, t)
+    return PlotData1D(solver(t), solution.semi)
 end
 
 #2D version of plot
-function dataplot(solution::DiscSolution{<:TwoDimension}, t)
-    return PlotData2D(solution.sol(t), solution.semi)
+function dataplot(solution::DiscSolution{<:TwoDimension}, solver, t)
+    return PlotData2D(solver(t), solution.semi)
 end
 
 dataplot(solution::DiscSolution{<:AbstractDimension}, t) = error("Unknown simulation type: $(typeof(t))")
 
-function plotgif(solution, tmin, tmax, kwargs...)
+function plotgif(solution::DiscSolution, tmin, tmax)#, kwargs...)
     ts = range(tmin, tmax, 150)
+    solver = @time solution.sol
     frames = Plots.@animate for t in ts
-        pd = dataplot(solution, t)
+        pd = dataplot(solution, solver, t)
         plot(
             pd,
             label = Printf.@sprintf("t = %1.2f", t),
-            kwargs
+            #ylims = (-1, 1.1), 
+            legend = :topright
+            #kwargs
         )
     end
     gif(frames, "temp.gif", fps = 10) |> display
 end
 
-#function Base.show(io::IO, sol::DiscSolution)
-#    println(io, sol.sol, sol.semi)
-#end
\ No newline at end of file
+function Base.show(io::IO, sol::DiscSolution)
+    println(io, string(typeof(sol)))
+end
```

Comment:
This is good, but it might be more useful to pretty print something along the lines of
```
DiscSolution{BurgersEquation}
```

In the REPL, play about with `typeof` to see what information you can pull out of the type, and print what you think the user needs to know.